pkgs/by-name/af/affine/package.nix

```diff
@@ -35,212 +36,203 @@ let
-          (cd $HOME/.electron-prebuilt-zip-tmp && zip --recurse-paths - .) > $ELECTRON_FORGE_ELECTRON_ZIP_DIR/electron-v$ELECTRON_VERSION_IN_LOCKFILE-${nodePlatform}-${nodeArch}.zip
-          export ELECTRON_SKIP_BINARY_DOWNLOAD=1
-
-          runHook postConfigurePhase
-        '';
-      buildPhase = ''
-        runHook preBuild
-
-        # first build
-        yarn workspaces focus @affine/electron @affine/monorepo
-        CARGO_NET_OFFLINE=true yarn workspace @affine/native build
-        BUILD_TYPE=${buildType} SKIP_NX_CACHE=1 yarn workspace @affine/electron generate-assets
-
-        # second build
-        yarn config set nmMode classic
-        yarn config set nmHoistingLimits workspaces
-        find . -name 'node_modules' -type d -prune -exec rm -rf '{}' +
-        yarn workspaces focus @affine/electron @affine/monorepo
-        BUILD_TYPE=${buildType} SKIP_WEB_BUILD=1 SKIP_BUNDLE=1 HOIST_NODE_MODULES=1 yarn workspace @affine/electron make
-
-        runHook postBuild
-      '';
-      installPhase =
-        let
-          inherit (finalAttrs) binName productName;
-        in
-        if hostPlatform.isDarwin then
-          ''
-            runHook preInstall
-
-            mkdir -p $out/Applications
-            mv packages/frontend/apps/electron/out/${buildType}/${productName}-darwin-${nodeArch}/${productName}.app $out/Applications
-
-            runHook postInstall
-          ''
-        else
-          ''
-            runHook preInstall
-
-            mkdir --parents $out/lib/${binName}/
-            mv packages/frontend/apps/electron/out/${buildType}/${productName}-linux-${nodeArch}/{resources,LICENSE*} $out/lib/${binName}/
-            install -Dm644 packages/frontend/apps/electron/resources/icons/icon_${buildType}_64x64.png $out/share/icons/hicolor/64x64/apps/${binName}.png
-
-            makeWrapper "${electron}/bin/electron" $out/bin/${binName} \
-              --inherit-argv0 \
-              --add-flags $out/lib/${binName}/resources/app.asar \
-              --add-flags "\''${NIXOS_OZONE_WL:+\''${WAYLAND_DISPLAY:+--ozone-platform-hint=auto --enable-features=WaylandWindowDecorations --enable-wayland-ime=true}}" \
-              --add-flags ${lib.escapeShellArg commandLineArgs}
-
-            runHook postInstall
-          '';
-    }
-    // (lib.optionalAttrs hostPlatform.isLinux {
-      desktopItems =
-        let
-          inherit (finalAttrs) binName productName;
-        in
-        [
-          (makeDesktopItem {
-            name = binName;
-            desktopName = productName;
-            comment = "AFFiNE Desktop App";
-            exec = "${binName} %U";
-            terminal = false;
-            icon = binName;
-            startupWMClass = binName;
-            categories = [ "Utility" ];
-            mimeTypes = [ "x-scheme-handler/${binName}" ];
-          })
-        ];
-    })
-  )
-)
+  desktopItems =
+    let
+      inherit (finalAttrs) binName productName;
+    in
+    [
```

Comment:
We should dedupe this and move this close to the top.

Reply:
What does this mean 😲

Reply:
We can do a let in after the finalAttrs attr to not have it inherited in the function multiple times.

Reply:
Do you mean
```
stdenv.mkDerivation (
  finalAttrs:
  let
    inherit (finalAttrs) binName productName;
  in {}
  )
```
?

Reply:
@SuperSandro2000 Can you review it?